### PR TITLE
Use DataEntities for serialization

### PR DESCRIPTION
**Pull request by peterdemartini**

### CODE REVIEW

File: asset/teraslice_kafka_reader/index.js [hunk 1]
```diff
@@ -232,7 +234,13 @@ function newReader(context, opConfig) {
...
                             endingOffsets[message.partition] = message.offset + 1;
 
-                            slice.push(message.value);
+                            const metadata = {
+                                partition: message.partition,
+                                offset: message.offset,
+                                topic: message.topic,
+                            };
+
+                            slice.push(DataEntity.fromBuffer(message.value, opConfig, metadata));
```
Comment (kstaken) [inline asset/teraslice_kafka_reader/index.js]: Will this default to json?

File: asset/teraslice_kafka_reader/index.js [hunk 2]
```diff
@@ -488,9 +496,18 @@ function schema() {
     };
 }
 
+function crossValidation(job) {
```
Comment (kstaken) [inline asset/teraslice_kafka_reader/index.js]: This should be noted as something to be removed once we complete the transition.

File: asset/teraslice_kafka_reader/index.js (hunk 1 above)
Comment (peterdemartini) [reply asset/teraslice_kafka_reader/index.js]: It will

File: asset/teraslice_kafka_reader/index.js (hunk 2 above)
Comment (peterdemartini) [reply asset/teraslice_kafka_reader/index.js]: Sure

**Pull request merged**